demo-ui/app/uk/gov/ons/addressIndex/demoui/modules/VersionModule.scala

```diff
@@ -0,0 +1,57 @@
+package uk.gov.ons.addressIndex.demoui.modules
+
+import java.util.UUID
+
+import com.google.inject.{ImplementedBy, Inject, Singleton}
+import uk.gov.ons.addressIndex.demoui.client.AddressIndexClientInstance
+import uk.gov.ons.addressIndex.model.AddressIndexSearchRequest
+import uk.gov.ons.addressIndex.model.server.response.AddressBySearchResponseContainer
+import scala.concurrent.Await
+import scala.concurrent.duration._
+import scala.concurrent.{ExecutionContext, Future}
+import scala.language.implicitConversions
+
+@ImplementedBy(classOf[DemoUIAddressIndexVersionModule])
+trait VersionModule {
+  def apiVersion: String
+  def dataVersion: String
+}
+
+/**
+  * Singleton class to get the API version number and data epoch once only
+  * @param apiClient
+  */
+@Singleton
+class DemoUIAddressIndexVersionModule @Inject()(
```

Comment:
This is a very peculiar way of getting a version.
What I thought was more in the spirit of extracting it from api response while getting a normal response from the api (by input, by uprn, bulk).
If the version needs to be display on every page (thus in the main template), I would opt for a creation of a separate api endpoint `/version` that will return just that.